### PR TITLE
feat(credential_store): add GOOGLE_WORKSPACE_CLI_KEYRING_BACKEND env var

### LINKED ISSUE #344: credential_store: .encryption_key file persists on disk even when OS keyring succeeds
## Bug

`get_or_create_key()` in `src/credential_store.rs` unconditionally writes the encryption key to `~/.config/gws/.encryption_key` on first run, **before** attempting to store it in the OS keyring. The file is never cleaned up after a successful keyring store.

This means on a macOS machine with a fully functional Keychain, the encryption key exists in **two** places:
1. macOS Keychain (protected by login credentials / biometrics)
2. `~/.config/gws/.encryption_key` (protected only by file permissions `0600`)

### Impact

The file-based key makes credential files portable — copying the entire `~/.config/gws/` directory to another machine transfers both the encrypted credentials and the key needed to decrypt them, defeating the purpose of keyring-based key storage.

### Root Cause

In `credential_store.rs` lines 64–120, the `Err(keyring::Error::NoEntry)` branch:

1. Generates a random 256-bit key
2. **Writes it to `.encryption_key`** (lines 102–115)
3. Then does best-effort `entry.set_password()` (line 118)

The file write happens unconditionally regardless of whether the keyring store succeeds.

### Proposed Fix

Only write `.encryption_key` as a fallback when the keyring is unavailable:

```rust
// Generate key
let mut key = [0u8; 32];
rand::thread_rng().fill_bytes(&mut key);
let b64_key = STANDARD.encode(key);

// Try keyring first
if entry.set_password(&b64_key).is_err() {
    // Keyring failed — persist to file as fallback
    write_key_file(&key_file, &b64_key);
} else {
    // Keyring succeeded — clean up any stale file
    let _ = std::fs::remove_file(&key_file);
}
```

### Verification

```bash
ls -la ~/.config/gws/.encryption_key
# Should NOT exist on systems with a working keyring
```

### PR DESCRIPTION
**Pull request by jpoehnelt**
## Summary

Adds gogcli-style backend selection for encryption key storage, inspired by [gogcli](https://github.com/steipete/gogcli).

### `GOOGLE_WORKSPACE_CLI_KEYRING_BACKEND`

| Value | Behavior |
|---|---|
| `keyring` (default) | OS keyring with `.encryption_key` file fallback |
| `file` | `.encryption_key` file only — for Docker/CI/headless |

### Key changes

- **Never delete `.encryption_key`** — it always serves as a durable fallback for ephemeral keyring environments
- When generating new keys with `backend=keyring`, save to **both** keyring and file
- Extract `KeyringProvider` trait + `resolve_key()` for testability
- 25 unit tests covering both backends and all edge cases
- Updated env var help, README, AGENTS.md

### Root cause (#344)

The previous fix for #344 deleted `.encryption_key` when the OS keyring succeeded. In Docker containers, the keyring is ephemeral, so the key was permanently lost on container restart. This PR preserves the file and adds an explicit env var to bypass the keyring entirely.

Fixes #344

### CODE REVIEW

**Comment by olvgrolle** [comment]
@jpoehnelt - cargo.toml is missing options to actually activate the actual keyring saves on macos / linux / pc: 

proposed diff: 

--- a/Cargo.toml
  +++ b/Cargo.toml
  @@ -66,3 +66,12 @@
   [dev-dependencies]
   serial_test = "3.4.0"
   tempfile = "3"
  +
  +[target.'cfg(target_os = "macos")'.dependencies]
  +keyring = { version = "3.6.3", features = ["apple-native"] }
  +
  +[target.'cfg(target_os = "linux")'.dependencies]
  +keyring = { version = "3.6.3", features = ["linux-native"] }
  +
  +[target.'cfg(target_os = "windows")'.dependencies]
  +keyring = { version = "3.6.3", features = ["windows-native"] }
